src/rect.rs

```diff
@@ -215,13 +215,13 @@ where
     T: Copy + PartialOrd + Add<T, Output = T> + Sub<T, Output = T>,
 {
     #[inline]
-    pub fn intersection(&self, other: &Self) -> Option<Self> {
+    pub fn try_intersection(&self, other: &Self) -> Option<NonEmpty<Self>> {
```

Comment:
I don't think `try_` returning `Option` is idiomatic. Perhaps, we could call it just `intersect`?